zerver/tests/test_signup.py

```diff
@@ -843,6 +854,29 @@ def test_successful_invite_user_as_guest_from_admin_account(self) -> None:
         self.assertFalse(invitee_profile.is_realm_admin)
         self.assertTrue(invitee_profile.is_guest)
 
+    def test_successful_invite_with_support_param(self) -> None:
+        invitor = self.example_user('iago')
+        invitor.is_staff = True
+        self.login(self.example_email('iago'))
+
+        invitee = self.nonreg_email('alice')
+        self.invite(invitee, ["Denmark"],
+                    invite_as=PreregistrationUser.INVITE_AS['GUEST_USER'])
+
+        prereg_user = PreregistrationUser.objects.get(email=invitee)
+        url = self.get_confirmation_url_from_outbox(invitee)
+        url += "?support=True"
+
+        self.assertEqual(prereg_user.status, 0)
+        response = self.client_get(url)
+        prereg_user.refresh_from_db()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(prereg_user.status, 0)
+
+        self.submit_reg_form_for_user(invitee, "password")
+        prereg_user.refresh_from_db()
+        self.assertEqual(prereg_user.status, 0)
```

Comment:
I would also add a test to verify that the user has been created after submitting the registration form.